src/index/VocabularyMergerImpl.h

```diff
@@ -181,7 +182,8 @@ CPP_template_def(typename C, typename L)(
       if (nextWord.isBlankNode()) {
         lastTripleComponent_->index_ = metaData_.getNextBlankNodeIndex();
       } else {
-        wordCallback(nextWord.iriOrLiteral(), nextWord.isExternal());
+        lastTripleComponent_->index_ =
+            wordCallback(nextWord.iriOrLiteral(), nextWord.isExternal());
```

Comment:
This whole block is confusing to read, as we sometimes use `nextWord` and sometimes `lastTripleComponent`. Can you make the `nextWork` a mutable reference and use it consistently in this index computation?